lib/utils/decodeText.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+
+let TextDecoder
+try {
+  TextDecoder = require('util').TextDecoder
+} catch (e) { }
+
+const { TextDecoder: PolyfillTextDecoder, getEncoding } = require('text-decoding')
+
+// Node has always utf-8
+const textDecoders = new Map()
+if (TextDecoder) {
+  textDecoders.set('utf-8', new TextDecoder('utf-8'))
+} else {
+  textDecoders.set('utf-8', new PolyfillTextDecoder('utf-8'))
+}
+textDecoders.set('utf8', textDecoders.get('utf-8'))
+
+function decodeText (text, textEncoding, destEncoding) {
+  if (text) {
+    if (textDecoders.has(destEncoding)) {
+      try {
+        return textDecoders.get(destEncoding).decode(Buffer.from(text, textEncoding))
+      } catch (e) { }
+    } else {
+      try {
+        textDecoders.set(destEncoding, new TextDecoder(destEncoding))
```

Comment:
@kibertoad 
here

Reply:
won't this be undefined in envs without ('util').TextDecoder?

Reply:
It is in a try catch block and second of all see line 60.

Reply:
ah, right, that makes perfect sense :D